atc/db/volume_repository.go

```diff
@@ -465,6 +468,58 @@ func (repository *volumeRepository) FindResourceCacheVolume(workerName string, r
 	return createdVolume, true, nil
 }
 
+func (repository *volumeRepository) FindWorkersForResourceCache(resourceCache ResourceCache) ([]string, error) {
+	rows, err := psql.Select("distinct(c.worker_name)").
```

Comment:
Hmm I think this will need some changes because it disregards the workers that are passed into the `Order(` method in the `placement.go` file. The list of workers that are passed into that method are a list of workers that are "compatible" (ex, non-team workers, tagged workers, etc) https://github.com/concourse/concourse/blob/dbbed7a7875441fe01784e05e485edb2079779c9/atc/worker/pool.go#L234-L237

Reply:
Good cache.

Reply:
Oh I was actually trying to describe that there is a problem that the list of workers that we query might not be a compatible worker and we were incrementing the `count[worker]++` in the map even if the worker is not a compatible worker. But I noticed you fixed it in your latest commit with https://github.com/concourse/concourse/blob/332618afd26014f612ecf199236e7201d9a16181/atc/worker/placement.go#L187 so great job!

Reply:
@clarafu As mentioned in https://github.com/concourse/concourse/pull/8063#discussion_r806320698, incrementing the `count[worker]++` was actually not a bug, because sort is only done on candidate workers, and `count` map only provides supporting data to the sort.

Reply:
Oh I'm sorry I didn't read that part properly earlier, thanks a lot for clarifying it to me again!